dot/node_test.go

```diff
@@ -0,0 +1,437 @@
+				}
+				// create trie from genesis
+				trie, err := genesis.NewTrieFromGenesis(gen)
+				if err != nil {
+					return nil, fmt.Errorf("failed to create trie from genesis: %w", err)
+				}
+				// create genesis block from trie
+				header, err := genesis.NewGenesisBlockFromTrie(trie)
+				if err != nil {
+					return nil, fmt.Errorf("failed to create genesis block from trie: %w", err)
+				}
+				stateSrvc.Telemetry = mockTelemetryClient
+				err = stateSrvc.Initialise(gen, header, trie)
+				if err != nil {
+					return nil, fmt.Errorf("failed to initialise state service: %s", err)
+				}
+
+				err = stateSrvc.SetupBase()
+				if err != nil {
+					return nil, fmt.Errorf("cannot setup base: %w", err)
+				}
+				return stateSrvc, nil
+			})
+
+			m.EXPECT().createRuntimeStorage(gomock.AssignableToTypeOf(&state.Service{})).Return(&runtime.
+				NodeStorage{}, nil)
+			m.EXPECT().loadRuntime(tt.args.cfg, &runtime.NodeStorage{}, gomock.AssignableToTypeOf(&state.Service{}),
+				ks, gomock.AssignableToTypeOf(&network.Service{})).Return(nil)
+			m.EXPECT().createBlockVerifier(gomock.AssignableToTypeOf(&state.Service{})).Return(&babe.
+				VerificationManager{}, nil)
+			m.EXPECT().createDigestHandler(log.Critical, gomock.AssignableToTypeOf(&state.Service{})).Return(&digest.
+				Handler{}, nil)
+			m.EXPECT().createCoreService(tt.args.cfg, ks, gomock.AssignableToTypeOf(&state.Service{}),
+				gomock.AssignableToTypeOf(&network.Service{}), &digest.Handler{}).Return(&core.Service{}, nil)
+			m.EXPECT().createGRANDPAService(tt.args.cfg, gomock.AssignableToTypeOf(&state.Service{}),
+				&digest.Handler{}, ks.Gran, gomock.AssignableToTypeOf(&network.Service{}),
+				gomock.AssignableToTypeOf(&telemetry.Mailer{})).Return(&grandpa.Service{}, nil)
+			m.EXPECT().newSyncService(tt.args.cfg, gomock.AssignableToTypeOf(&state.Service{}), &grandpa.Service{},
+				&babe.VerificationManager{}, &core.Service{}, gomock.AssignableToTypeOf(&network.Service{}),
+				gomock.AssignableToTypeOf(&telemetry.Mailer{})).Return(&gsync.Service{}, nil)
+			m.EXPECT().createBABEService(tt.args.cfg, gomock.AssignableToTypeOf(&state.Service{}), ks.Babe,
+				&core.Service{}, gomock.AssignableToTypeOf(&telemetry.Mailer{})).Return(&babe.Service{}, nil)
+			m.EXPECT().createSystemService(&tt.args.cfg.System, gomock.AssignableToTypeOf(&state.Service{})).
+				DoAndReturn(func(cfg *types.SystemInfo, stateSrvc *state.Service) (*system.Service, error) {
+					gd, err := stateSrvc.Base.LoadGenesisData()
+					return system.NewService(cfg, gd), err
+				})
+			m.EXPECT().createNetworkService(tt.args.cfg, gomock.AssignableToTypeOf(&state.Service{}),
+				gomock.AssignableToTypeOf(&telemetry.Mailer{})).Return(testNetwork, nil)
+
+			got, err := newNode(tt.args.cfg, ks, m)
+			if tt.err != nil {
+				assert.EqualError(t, err, tt.err.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+
+			if tt.want != nil {
+				assert.NotNil(t, got)
+			} else {
+				assert.Nil(t, got)
+			}
```

Comment:
```suggestion
			assert.Equal(tm tt.want, got)
```

Reply:
Added assert.Equal for check, not checking whole struct because checking assert.Equal on node.Services causes a SIGABRT.